.ci/validation.sh

```diff
@@ -53,9 +53,7 @@ check-missing-pitests)
   for item in "${list[@]}"
   do
     item=${item//\./\/}
-    if [[ $item == */\*  ]] ; then
-     item=$item
-    else
+    if [[ $item != */\*  ]] ; then
       if [[ $item != *\* ]] ; then
```

Comment:
Am I blind? I see line 56 and 57 are same.

Reply:
Are they really? I am still learning shell and was not sure of that one. This seems like a partial string comparison but is it checking for the characters `/` and `\` or is it escaping characters or what is happening? @Rahulkhinchi03 can you tell what you think?

Reply:
It was late time review , now I see a difference.

Reply:
How can we test this update?
She'll script changes are not easy to confirm analyticaly

Reply:
I just added `echo "${item}" >> ~/Desktop/master.txt` after the initial if-statement in the `master` branch and once in this pull request's branch - `echo "${item}" >> ~/Desktop/pull_request.txt`. There seems to be no `diff`erences. Is this a legit confirmation in your opinion?

```console
stoyan@ZenBook:~/Desktop$ ls *.txt
master.txt  pull_request.txt
stoyan@ZenBook:~/Desktop$ diff master.txt pull_request.txt 
stoyan@ZenBook:~/Desktop$ cat master.txt 
com/puppycrawl/tools/checkstyle/meta/*
com/puppycrawl/tools/checkstyle/checks/ArrayTypeStyleCheck.java
com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java
com/puppycrawl/tools/checkstyle/checks/DescendantTokenCheck.java
com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java
com/puppycrawl/tools/checkstyle/checks/LineSeparatorOption.java
com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheck.java
com/puppycrawl/tools/checkstyle/checks/NoCodeInFileCheck.java
com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java
com/puppycrawl/tools/checkstyle/checks/OuterTypeFilenameCheck.java
com/puppycrawl/tools/checkstyle/checks/sizes/RecordComponentNumberCheck.java
com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java
com/puppycrawl/tools/checkstyle/checks/TodoCommentCheck.java
com/puppycrawl/tools/checkstyle/checks/TrailingCommentCheck.java
com/puppycrawl/tools/checkstyle/checks/TranslationCheck.java
com/puppycrawl/tools/checkstyle/checks/UncommentedMainCheck.java
com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheck.java
com/puppycrawl/tools/checkstyle/checks/UpperEllCheck.java
com/puppycrawl/tools/checkstyle/checks/annotation/*
com/puppycrawl/tools/checkstyle/checks/blocks/*
com/puppycrawl/tools/checkstyle/checks/coding/AbstractSuperCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/ArrayTrailingCommaCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/AvoidDoubleBraceInitializationCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/AvoidInlineConditionalsCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/AvoidNoArgumentSuperConstructorCallCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/CovariantEqualsCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/DeclarationOrderCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/EqualsHashCodeCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/IllegalCatchCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/IllegalThrowsCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/IllegalTokenCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/SuperCloneCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/SuperFinalizeCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/ReturnCountCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanExpressionCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/IllegalInstantiationCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/MagicNumberCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/MissingCtorCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/MissingSwitchDefaultCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/ModifiedControlVariableCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/MultipleStringLiteralsCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/MultipleVariableDeclarationsCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/NestedForDepthCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/NestedIfDepthCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/NestedTryDepthCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/IllegalTokenTextCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonInTryWithResourcesCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/OneStatementPerLineCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/NoArrayTrailingCommaCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/NoCloneCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/NoEnumTrailingCommaCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/NoFinalizerCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/StringLiteralEqualityCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/ExplicitInitializationCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterOuterTypeDeclarationCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonAfterTypeMemberDeclarationCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/UnnecessarySemicolonInEnumerationCheck.java
com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java
com/puppycrawl/tools/checkstyle/checks/design/*
com/puppycrawl/tools/checkstyle/checks/header/*
com/puppycrawl/tools/checkstyle/checks/imports/*
com/puppycrawl/tools/checkstyle/checks/indentation/*
com/puppycrawl/tools/checkstyle/checks/javadoc/*
com/puppycrawl/tools/checkstyle/checks/metrics/*
com/puppycrawl/tools/checkstyle/checks/modifier/*
com/puppycrawl/tools/checkstyle/checks/naming/*
com/puppycrawl/tools/checkstyle/checks/regexp/*
com/puppycrawl/tools/checkstyle/checks/sizes/*
com/puppycrawl/tools/checkstyle/checks/whitespace/*
com/puppycrawl/tools/checkstyle/ant/*
com/puppycrawl/tools/checkstyle/PackageNamesLoader.java
com/puppycrawl/tools/checkstyle/AuditEventDefaultFormatter.java
com/puppycrawl/tools/checkstyle/ConfigurationLoader.java
com/puppycrawl/tools/checkstyle/DefaultConfiguration.java
com/puppycrawl/tools/checkstyle/DefaultContext.java
com/puppycrawl/tools/checkstyle/DefaultLogger.java
com/puppycrawl/tools/checkstyle/Definitions.java
com/puppycrawl/tools/checkstyle/XMLLogger.java
com/puppycrawl/tools/checkstyle/SarifLogger.java
com/puppycrawl/tools/checkstyle/MetadataGeneratorLogger.java
com/puppycrawl/tools/checkstyle/PackageObjectFactory.java
com/puppycrawl/tools/checkstyle/PropertiesExpander.java
com/puppycrawl/tools/checkstyle/PropertyCacheFile.java
com/puppycrawl/tools/checkstyle/Checker.java
com/puppycrawl/tools/checkstyle/ThreadModeSettings.java
com/puppycrawl/tools/checkstyle/grammar/CrAwareLexerSimulator.java
com/puppycrawl/tools/checkstyle/AuditEventFormatter.java
com/puppycrawl/tools/checkstyle/XdocsPropertyType.java
com/puppycrawl/tools/checkstyle/PropertyType.java
com/puppycrawl/tools/checkstyle/FileStatefulCheck.java
com/puppycrawl/tools/checkstyle/ModuleFactory.java
com/puppycrawl/tools/checkstyle/PropertyResolver.java
com/puppycrawl/tools/checkstyle/StatelessCheck.java
com/puppycrawl/tools/checkstyle/TreeWalkerFilter.java
com/puppycrawl/tools/checkstyle/GlobalStatefulCheck.java
com/puppycrawl/tools/checkstyle/grammar/CommentListener.java
com/puppycrawl/tools/checkstyle/DetailAstImpl.java
com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java
com/puppycrawl/tools/checkstyle/XmlLoader.java
com/puppycrawl/tools/checkstyle/JavaParser.java
com/puppycrawl/tools/checkstyle/Main.java
com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java
com/puppycrawl/tools/checkstyle/DetailNodeTreeStringPrinter.java
com/puppycrawl/tools/checkstyle/AstTreeStringPrinter.java
com/puppycrawl/tools/checkstyle/SuppressionsStringPrinter.java
com/puppycrawl/tools/checkstyle/TreeWalker.java
com/puppycrawl/tools/checkstyle/TreeWalkerAuditEvent.java
com/puppycrawl/tools/checkstyle/CheckstyleParserErrorStrategy.java
com/puppycrawl/tools/checkstyle/api/*
com/puppycrawl/tools/checkstyle/filefilters/*
com/puppycrawl/tools/checkstyle/filters/*
com/puppycrawl/tools/checkstyle/utils/*
com/puppycrawl/tools/checkstyle/gui/*
com/puppycrawl/tools/checkstyle/xpath/*
com/puppycrawl/tools/checkstyle/XpathFileGeneratorAuditListener.java
com/puppycrawl/tools/checkstyle/XpathFileGeneratorAstFilter.java
com/puppycrawl/tools/checkstyle/JavaAstVisitor.java

```